### DISCUSSION TITLE
Is there a way to pass `None` to a non-optional field and make it use the default?

### DISCUSSION THREAD

**Discussion by jonathanhuang13** [pydantic/pydantic#7697] [Question] [answered]
I want to be able to do something like this:
```
class User(BaseModel):
    name: str = Field(default="John Doe")

user = User(name=None)
# user.name = "John Doe"
```

I know there's a way to do this using [validators](https://stackoverflow.com/questions/63616798/how-to-pass-the-default-value-to-a-variable-if-none-was-passed) but was curious if there is a way to do this without validators since nothing is actually being validated

**Comment by enzanumo** [comment]
I encountered the same problem once, and my workaround is to avoid passing None in the first place...

```py3
kw = dict(name=None)
user = User(**{k:v for k, v in kw.items() if v is not None})
```

**Comment by samuelcolvin** [comment, marked as answer]
It's definitely possible in pydantic-core, see [here](https://github.com/pydantic/pydantic-core/blob/d93482e3b83f33cc134e8aa83992467c6dda8395/python/pydantic_core/core_schema.py#L2172).

I'm not sure if we expose this in pydantic yet, but you could achieve it with a custom type.

Although this isn't exactly "without validation", it's just using the default when validation fails.

**Reply by samuelcolvin** [reply]
If we don't expose this already (afaik we don't), PR welcome to add it as an argument to `Field`
